File: server/php/src/Application/Services/HostedTokenizationService.php
<?php

namespace OnlinePayments\ExampleApp\Application\Services;

use Exception;
use OnlinePayments\ExampleApp\Application\DTOs\HostedTokenization\ResponseDto;
use OnlinePayments\ExampleApp\Application\Interfaces\SDKClients\HostedTokenizationClientInterface;
use OnlinePayments\ExampleApp\Application\Interfaces\Services\HostedTokenizationServiceInterface;

readonly class HostedTokenizationService implements HostedTokenizationServiceInterface
{
    public function __construct(
        private HostedTokenizationClientInterface $hostedTokenizationClient,
    ) {}

    /**
     * @throws Exception
     */
    public function initHostedTokenization(): ResponseDto
    {
        return $this->hostedTokenizationClient->initHostedTokenization();
    }
}

File: server/python/app/application/services/hosted_tokenization_service.py
from app.application.dtos.hosted_tokenization.response_dto import HostedTokenizationResponseDto
from app.application.interfaces.sdk_clients.hosted_tokenization_client_interface import HostedTokenizationClientInterface
from app.application.interfaces.services.hosted_tokenization_service_interface import HostedTokenizationServiceInterface

class HostedTokenizationService(HostedTokenizationServiceInterface):

    def __init__(self, hosted_tokenization_client: HostedTokenizationClientInterface):
        self._client = hosted_tokenization_client

    def init_hosted_tokenization(self) -> HostedTokenizationResponseDto:
        return self._client.init_hosted_tokenization()